devicetypes/qubino/qubino-dimmer.src/qubino-dimmer.groovy

```diff
@@ -0,0 +1,634 @@
+			} else {
+				runIn(5, "syncConfiguration", [overwrite: true])
+			}
+		} else {
+			state.currentPreferencesState."$key"?.status = "syncPending"
+			runIn(5, "syncConfiguration", [overwrite: true])
+		}
+	}
+	// Preferences template end
+}
+
+private getPreferenceValue(preference, value = "default") {
+	def integerValue = value == "default" ? preference.defaultValue : value.intValue()
+	switch (preference.type) {
+		case "enum":
+			return String.valueOf(integerValue)
+		case "boolean":
+			return String.valueOf(preference.optionActive == integerValue)
+		default:
+			return integerValue
+	}
+}
+
+private getCommandValue(preference) {
+	def parameterKey = preference.key
+	switch (preference.type) {
+		case "boolean":
+			return settings."$parameterKey" ? preference.optionActive : preference.optionInactive
+		case "boolRange":
+			def parameterKeyBoolean = parameterKey + "Boolean"
+			return !notNullCheck(settings."$parameterKeyBoolean") || settings."$parameterKeyBoolean" ? settings."$parameterKey" : preference.disableValue
+		case "range":
+			return settings."$parameterKey"
+		default:
+			return Integer.parseInt(settings."$parameterKey")
+	}
+}
+
+private notNullCheck(value) {
+	return value != null
+}
+
+private isPreferenceChanged(preference) {
+	if (notNullCheck(settings."$preference.key")) {
+		if (preference.type == "boolRange") {
+			def boolName = preference.key + "Boolean"
+			if (state.currentPreferencesState."$preference.key".status == "disabled") {
+				return settings."$boolName"
+			} else {
+				return state.currentPreferencesState."$preference.key".value != settings."$preference.key" || !settings."$boolName"
+			}
+		} else {
+			return state.currentPreferencesState."$preference.key".value != settings."$preference.key"
+		}
+	} else {
+		return false
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.multichannelv3.MultiChannelCmdEncap cmd, ep = null) {
+	log.debug "Multichannel command ${cmd}" + (ep ? " from endpoint $ep" : "")
+	def encapsulatedCommand = cmd.encapsulatedCommand()
+	zwaveEvent(encapsulatedCommand, cmd.sourceEndPoint as Integer)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.multichannelv3.MultiChannelEndPointReport cmd, ep = null) {
+	log.info "MultiChannelEndPointReport: ${cmd}"
+	response([
+			refresh()
+	])
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd) {
+	// Handles other Z-Wave commands that are not supported here
+	log.debug "Command: ${cmd}"
+	[:]
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd, ep = null) {
+	log.debug "BasicReport: ${cmd}"
+	dimmerEvents(cmd)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicSet cmd, ep = null) {
+	log.debug "BasicSet: ${cmd}"
+	def input1SwitchType = Integer.parseInt(state.currentPreferencesState.input1SwitchType.value)
+
+	if(input1SwitchType == INPUT_TYPE_POTENTIOMETER) {
+		log.debug "BasicSet: ${cmd} / INPUT_TYPE_POTENTfIOMETER"
+		[response(zwave.switchMultilevelV3.switchMultilevelGet().format())]
```

Comment:
you don't need to call `.format()` here, `response()` accepts zwave command objects.

Also why is it wrapped in brackets?

Reply:
yes, wrapping was unnecessary. Fixed.